utility_analysis_new/dp_engine.py

```diff
@@ -50,11 +51,13 @@ def aggregate(self,
                          "If you like to perform DP computations use "
                          "DPEngine.aggregate.")
 
-    def analyze(self,
-                col,
-                options: utility_analysis_new.UtilityAnalysisOptions,
-                data_extractors: pipeline_dp.DataExtractors,
-                public_partitions=None):
+    def analyze(
+            self,
+            col,
+            options: utility_analysis_new.UtilityAnalysisOptions,
+            data_extractors: Union[pipeline_dp.DataExtractors,
```

Comment:
If we indeed need a dedicated data structure for PreAggregatedExtractors, please  could you update the function doc and explaining that PreAggregateExtractors should be provided when utility analysis is run on pre-aggregated data, otherwise DataExtractors should be provided.